xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -240,17 +238,7 @@ func (c *clusterNode) handleResp(clusterUpdate xdsclient.ClusterUpdate, err erro
 	// possibility for a full cluster update to successfully build, as those created children will not have received
 	// an update yet. However, if there was simply a child deleted, then there is a possibility that it will have a full
 	// cluster update to build and also will have a changed overall cluster update from the deleted child.
-	if clusterUpdate.ClusterType == xdsclient.ClusterTypeAggregate {
-		if deletedChild && len(newChildren) == 0 {
-			c.clusterHandler.constructClusterUpdate()
-			return
-		}
-	}
-
-	// If the cluster was a leaf node, if the cluster update received had change in the cluster update then the overall
-	// cluster update would change and there is a possibility for the overall update to build so ping cluster handler to
-	// return.
-	if deltaInChildUpdateFields {
+	if (deletedChild && !createdChild) || deltaInClusterUpdateFields {
```

Comment:
Why did you also `|| deltaInClusterUpdateFields`?

If it's true, (and since cluster name never changes), this is a type change from leaf to aggregated. And in that case, we don't want to send update, right?

I think it's OK to only check for `deletedChild && !createdChild`

Reply:
I assumed clustername could change from the wire from traffic director.

Reply:
Deleted or.